core/database_util.go

```diff
@@ -164,6 +95,20 @@ func GetHeadFastBlockHash(db ethdb.Database) common.Hash {
 	return common.BytesToHash(data)
 }
 
+// GetHeadBlockNum retrieves the block number of the current canonical head block.
+func GetHeadBlockNum(db ethdb.Database) *big.Int {
```

Comment:
All the methods currently in this file have been individually tested and validated, this one was not. Please add a test, that should have caught the issue.